### ISSUE TITLE
Upstream high latency

### ISSUE THREAD

**Issue by aminvakil** [vozlt/nginx-module-vts#222]
Something between latest tag now which is https://github.com/vozlt/nginx-module-vts/releases/tag/v0.1.18 and latest commit now which is https://github.com/vozlt/nginx-module-vts/commit/3c6cf4 causes vts to report that upstream requests response slower.

I have seen this 6 months ago and switched to latest tag and everything was fixed, sorry for opening this issue so late.

I can't remember completely if it was just vts reporting incorrectly or upstream responses got slower too. We were changing many different things and as quickly as we saw sticking to latest tag (v0.1.18) fixes our problem, we didn't change that.

I will try bumping, compiling, building with newer commits than (v0.1.18 tag), but that takes a lot of time.

But let's put this here and see if someone else had the same experience or someone is willing to help.

Thank you for creating and maintaining this!

**Comment by testn** [comment]
Can you try to bisect and see which one caused the problem?

**Comment by aminvakil** [comment]
Yeah, I probably should do this at some time, but this takes a lot of time as we cannot test this on our production systems, I should have do some load-testing with each build too.

**Comment by u5surf** [comment]
There is a change in that range that matches what you saw: #145, merged January 2021, "Fixed incorrect calculation of nginx_vts_upstream_request_seconds and nginx_vts_upstream_response_seconds values".

The average divided the sum of the samples inside the period by the size of the queue rather than by the number of samples it had actually added:

```diff
-return (ngx_msec_t) (k / (q->len - 1));
+return (c == 0) ? (ngx_msec_t) 0 : (ngx_msec_t) (k / c);
```

With the default `AMM 60s`, a zone taking ten requests a minute was reported at about a sixth of its real average before that commit and correctly after it; a zone busy enough to fill the 63 slots within the period was already right. So v0.1.18 was under-reporting rather than the newer build over-reporting, and the upstreams themselves were not slower.

Closing, since there is no reproduction here and the thread has been quiet since 2022. Please reopen with a configuration and the numbers if the reported time still disagrees with `$upstream_response_time` in the access log for the same requests.
